syntax = "proto3"; //声明protobuf的版本

package fixbug; //声明代码所在的包（对于C++来说是namespace）

message ResultCode {
  int32 errcode = 1;
  bytes errmg = 2;
}
message LoginRequest {
  bytes name = 1;
  bytes pwd = 2;
}

message LoginResponse {
  ResultCode result_code = 1;
  bool success = 3;
}

message GetFriendListsRequest {
  uint32 userid = 1;
}

message User {
  bytes name = 1;
  uint32 age = 2;
  enum Sex {
    MAN = 0;
    WOMAN = 1;
  }
  Sex sex = 3;
}

message GetFriendListsResponse {
  ResultCode result_code = 1;
  repeated User friend_list = 2; //定义一个列表类型
}